### ISSUE TITLE
Improve API Failure Error Message

### ISSUE THREAD

**Issue by easyaspi314** [Headline/discord-compiler-bot#21]
As indicated by the failed tests, the latest commit leaves the bot in an unusable state:
```
Wandbox request failure 
 FetchError: invalid json response body
at https://wandbox.org/api/compile.json

reason: Unexpected token < in JSON at position 0 
Please try again later
```

**Comment by easyaspi314** [comment]
Wait, it seems that wandbox.org itself is down:
![Screenshot_20200411-123140](https://user-images.githubusercontent.com/6258309/79049248-798ef180-7bf0-11ea-98c5-18b00e47e7db.png)

**Comment by easyaspi314** [comment]
Talked to Wandbox's maintainer, and they brought it back up.

**Comment by Headline** [comment]
Thanks for creating this issue! I would be never realized

How did the bot behave?

**Comment by easyaspi314** [comment]
It just hung and then posted that error message.
![Screenshot_20200411-160856](https://user-images.githubusercontent.com/6258309/79053894-c7b2ed80-7c0e-11ea-9506-99a1d2d118b6.png)

**Comment by easyaspi314** [comment]
I presume the API is sending said 504 message in HTML format, and that is it trying to parse the doctype declaration.

**Comment by Headline** [comment]
Yeah that sounds about right

I’ll reopen this so I can take a look at what checks we have here and hopefully improve that error

**Comment by easyaspi314** [comment]
According to node-fetch's [README.md](https://github.com/node-fetch/node-fetch/blob/master/README.md#handling-client-and-server-errors), you should be able to do something like this:
```js

const fetch = require('node-fetch');

function checkStatus(res) {
	if (res.ok) {
		// res.status >= 200 && res.status < 300
		return res;
	} else {
		throw MyCustomError(res.statusText);
	}
}

fetch('https://httpbin.org/status/400')
	.then(checkStatus)
	.then(res => console.log('will not get here...'));
```

**Comment by Headline** [comment]
This has been resolved in 441fb39, thanks!
